lib/ansible/modules/system/known_hosts.py

```diff
@@ -119,12 +119,15 @@ def enforce_state(module, params):
 
     found,replace_or_add,found_line,key=search_for_host_key(module,host,key,hash_host,path,sshkeygen)
 
+    params['diff'] = compute_diff(path, found_line, replace_or_add, state, key)
```

Comment:
This function returns `params`, which is then passed to `module.exit_json()`.  I need to set `params['diff']` to make the diff possible when `check_mode` is `False`.